### ISSUE TITLE
Cache::Async

### ISSUE THREAD

**Issue by AlexDaniel** [perl6/ecosystem-unbitrot#574]
Module [Cache::Async](https://modules.perl6.org/dist/Cache::Async) cannot be installed (`Flapper`), perhaps it has some failing tests.

* Tickets are opened/closed in this repo automatically (though not immediately).
* If you can **install the module without any problems**, add `works for me` label, leave a comment saying that it works for you and mention any details that you feel are important.
* If it **needs a native library**, put `native dependency` label, describe what you did to install it and ensure that same instructions are present in the README file of the module (otherwise submit a pull request). Also try to update [this wiki page](https://github.com/perl6/ecosystem-unbitrot/wiki).
* If the **module is broken**, try to fix it and send a PR. Add `PR sent` label.
* If there is a **problem in one of the dependencies**, add `failing dependency` label and write a comment explaining the situation. Feel free to work on the corresponding ticket for the failing dependency.
* It is a good idea to **assign yourself** to this ticket if you're working on it (to make sure two or more people are not working on the same ticket at the same time).
* **Once you are done**, [search for a next ticket](https://github.com/perl6/ecosystem-unbitrot/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+-label%3A%22PR+sent%22+-label%3A%22deprecated+module%22+-label%3A%22native+dependency%22).

If you can't self-assign yourself or attach a label, please let us know on [#perl6 channel on freenode](https://perl6.org/irc) or just leave a comment here. We will try to give you privileges as fast as possible.

Output:

````````````````````````````````````````````````````````
===> Searching for: Cache::Async
===> Found: Cache::Async:ver<0.1.3> [via Zef::Repository::Ecosystems<cpan>]
===> Fetching [OK]: Cache::Async:ver<0.1.3> to /home/alex/Blin/data/zef-data/tmp/1551238471.5427.5232/perl6-cache-async-0.1.3.tar.gz
===> Extraction [OK]: Cache::Async to /home/alex/Blin/data/zef-data/store/perl6-cache-async-0.1.3.tar.gz
===> Testing: Cache::Async:ver<0.1.3>
t/00-meta.t ............ ok
t/01-basics.t .......... ok
t/02-async.t ........... ok
# Failed test 'pausing between gets evicts the older entries'
# at t/03-expiry.t line 90
# expected: 'ABCDBA'
#      got: 'ABCDCBA'
# Looks like you failed 1 test of 12
t/03-expiry.t .......... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/12 subtests 
t/04-exceptions.t ...... ok
t/05-monitoring.t ...... ok
t/06-slurpies.t ........ ok
t/07-async-producer.t .. ok
t/08-refreshing.t ...... ok

Test Summary Report
-------------------
t/03-expiry.t        (Wstat: 256 Tests: 12 Failed: 1)
  Failed test:  12
  Non-zero exit status: 1
Files=9, Tests=63, 19 wallclock secs ( 0.07 usr  0.02 sys + 14.53 cusr  0.84 csys = 15.46 CPU)
Result: FAIL
===> Testing [FAIL]: Cache::Async:ver<0.1.3>
Failed to get passing tests, but continuing with --force-test
===> Installing: Cache::Async:ver<0.1.3>
===> Install [OK] for Cache::Async:ver<0.1.3>

````````````````````````````````````````````````````````

Ping @robertlemmen

Previous ticket: #556

**Comment by AlexDaniel** [comment]
Issue https://github.com/robertlemmen/perl6-cache-async/issues/1

> I'll close the issue for now, but if anyone sees these flapping again then please let me know and I'll try to tune them further.

@robertlemmen well, they seem to be flapping :)

**Comment by robertlemmen** [comment]
ah darn. I don't really have a good strategy to test these inherently time-sensitive things...

**Comment by AlexDaniel** [comment]
```
sleep .06;
```
That's a really small amount to sleep for. Can you bump that to say 0.25s? That would take around 2 seconds to complete, but that's gonna work every single time. If you're worried about your tests taking way too long, parallelize them.

**Comment by robertlemmen** [comment]
yeah, that's what I am going to do. I am  not reallly worried about overall test execution time yet

**Comment by Altai-man** [comment]
Using https://github.com/FCO/test-time should help. Alternatively, https://perl6advent.wordpress.com/2016/12/17/testing-in-virtual-time/ <- here is the original module, I think.

**Comment by robertlemmen** [comment]
I have made a version 0.1.6 with less strict timings, and also made the 09-jitter test run multiple times and work an average. if this is not enough to make the tests reliable, I'll try test-time, which looks really awesome

**Comment by robertlemmen** [comment]
is this still an issue with the recent changes to cache::async? if so I would have another stab at it as discussed above, if not then I would like to close the other issue as well...

**Comment by AlexDaniel** [comment]
@robertlemmen feel free. If it's still an issue, a new ticket will be created automatically. If it's resolved, this ticket will be closed automatically if you don't close it yourself.
